lib/util/types.ts

```diff
@@ -19,8 +19,13 @@ export interface Request {
   delete (options: RequestOptions): Promise<any>
 }
 
-export default interface TaxjarError extends Error {
-  error: string,
-  detail: string,
-  status: number
+export class TaxjarError extends Error {
+  constructor(
+    public error: string,
+    public detail: string,
+    public status: number
+  ) {
+    super(`TaxJar: ${error} - ${detail}`);
+    Object.setPrototypeOf(this, TaxjarError.prototype);
```

Comment:
When extending an error it is.
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-2.html#support-for-newtarget

Though it seems there is a newer syntax:
```suggestion
    Object.setPrototypeOf(this, new.target.prototype);
```

Reply:
@squirly I believe that only applies when you're transpiling to ES5. Since we're still using ES6 in published files, everything seems to work without calling `Object.setPrototypeOf`.

Reply:
Makes sense.

```suggestion
```